device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/IotHubConnectionStateCallback.java

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.azure.sdk.iot.device;
+
+/**
+ * An interface for an IoT Hub connection state callback.
+ *
+ * Developers are expected to create an implementation of this interface,
+ * and the transport will call {@link IotHubConnectionStateCallback#connectionUp()}
+ * when connection is established with the IoT Hub and call
+ * {@link IotHubConnectionStateCallback#connectionDown()} when connection is lost.
+ */
+public interface IotHubConnectionStateCallback {
+    /**
+     * Method executed when the connection with the IoTHub is lost.
+     */
+    void connectionUp();
```

Comment:
To keep this aligned across various SDK's we have, It would be nice to replace this with enum connectionStatus {
CONNECTION_SUCCESS, CONNECTION_DROP } and provide the enum as status as opposed to having multiple methods. This way we can easily extend this enum to multiple states associated with connection (for eg - in future we can add retry in progress, retry time out etc etc)